extensions/wikia/Recirculation/js/helpers/recommendedContent.js

```diff
@@ -0,0 +1,97 @@
+/*global define*/
+define('ext.wikia.recirculation.helpers.recommendedContent', [
+    'jquery',
+    'wikia.window',
+    'wikia.nirvana',
+    'ext.wikia.recirculation.helpers.blacklist',
+    'wikia.log'
+], function ($, w, nirvana, blacklist, log) {
+    'use strict';
+
+    var numberOfArticleFooterSlots = $('#mixed-content-footer').data('number-of-wiki-articles');
+    var REQUEST_ID;
+
+    function getFilteredItems(response) {
+        var blacklistedItems = blacklist.get();
+
+        return response.article_recommendation
+            .concat(response.wiki_recommendation)
+            .filter(function (el) {
+                return blacklistedItems.indexOf(el.item_id) === -1;
+            });
+    }
+
+    function mapExperimentalDataResponse(response) {
+        var filteredItems = getFilteredItems(response);
+
+        REQUEST_ID = response.recommendation_request_id;
+
+        if (filteredItems < numberOfArticleFooterSlots) {
+            blacklist.remove(5);
+
+            filteredItems = getFilteredItems(response);
+        }
+
+        return filteredItems
+            .map(function (el) {
+                return {
+                    id: el.item_id,
+                    site_name: el.wiki_title,
+                    url: el.url,
+                    thumbnail: el.thumbnail_url,
+                    title: el.article_title || el.wiki_title,
+                };
+            });
+    }
+
+    function getExperimentalRecommendedData() {
+        var deferred = $.Deferred();
+        var itemId = w.wgCityId + '_' + w.wgArticleId;
+
+        blacklist.update(itemId);
+
+        $.ajax({
+            url: w.wgServicesExternalDomain + 'recommendations/recommendations',
+            data: {
+                wikiId: w.wgCityId,
+                articleId: w.wgArticleId,
+                beacon: w.beacon_id,
+            }
+        }).done(function (result) {
+            deferred.resolve(mapExperimentalDataResponse(result));
+        }).fail(function (err) {
+            log('Failed to fetch experimental recommended data, using getPopularPages as backup' + err, log.levels.error);
```

Comment:
does this log method send data to event-logger? I think there was requirement about it

Reply:
I've added support for event logger here https://github.com/Wikia/app/pull/16934/files#diff-3d840aaade76d8d66ebe3fae6b6d0f66

and here https://github.com/Wikia/app/pull/16934/files#diff-3bdaa9c4c0484bb666c559d8cd1212e7R33
as they want to get that event in case of not enough items